### ISSUE TITLE
what is this problem and how to deal with？please

### ISSUE THREAD

**Issue by DSAJLDNA** [ZelongZeng/PLCD#2]
D:\Anaconda\envs\Ren\lib\site-packages\torchvision\models\_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'wei
ghts' instead.
  warnings.warn(
D:\Anaconda\envs\Ren\lib\site-packages\torchvision\models\_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be remov
ed in the future. The current behavior is equivalent to passing `weights=ResNet50_Weights.IMAGENET1K_V1`. You can also use `weights=ResNet50_Weights.DEFAULT` to get the most up-to-date we
ights.
  warnings.warn(msg)
Load the model from ./model\G2D\net_119_s0.pth
2 -> 3:
This problem comes up in cycles，what is this problem and how to deal with？please

**Comment by ZelongZeng** [comment]
Hi DSAJLDNA,
The warning arises from difference in loading pretrained models between the old and new versions of PyTorch. Generally speaking, this warning should not affect the code's execution.

If you want to address this issue, I suggest:

1. Use an older version of torchvision (<0.13).
2. Follow the warning message and modify the relevant parts of the code. 
Change the code on line 124 of file `model_rmac.py` from `model_ft = models.resnet50(pretrained=True) `to `model_ft = models.resnet50(pretrained=ResNet50_Weights.IMAGENET1K_V1)`. 

I hope these ways can solve your problem.

**Comment by DSAJLDNA** [comment]
Thank you very much for your answer, I now have a new issue when I'm running “python train_PL_G2D.py --train_all --batchsize 8 --pool max --data_dir D:\University\University-Release\train --gen 0 --seed 0 --name step1 --gpu 0 --lambda1 1.0 --lambda2 0.1 --easy_pos  ”，The following error was encountered with this string of code：
An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
Does this need to be changed in the source code, or does my computer not support multi-threading?

**Comment by ZelongZeng** [comment]
Hi,
I think the following step can solve the problem:
1. Define the section from line 419 to line 452 in the file `train_PL_G2D.py` as a function: `def run_main()`.
2. Add the following at the end of the file `train_PL_G2D.py`:
```
      if __name__ == '__main__':
         run_main()
```
I hope these ways can solve your problem. 
You can also google the error or `if __name__ == '__main__'` to get more information about this modification.

**Comment by DSAJLDNA** [comment]
Thank you very much for your answer, I have solved the second question, but when I debug the pre-trained model, the following loop appears, I would like to ask what is the situation. This is the order "python test_PL_G2D.py --batchsize 8 --test_dir D:\University\University-Release\test --name G2D --gpu_ids 0 --low_altitude ".
We use the scale: 1
17119
-------test-----------
Load the model from ./model\G2D\net_119_s0.pth
2 -> 3:

**Comment by ZelongZeng** [comment]
Hi, 
Sorry, I cannot quite understand your question clearly. Could you please describe it in more detail?

**Comment by DSAJLDNA** [comment]
It's okay, I've solved the above problem, but now I have a new problem, why does the training effect start from scratch every time I re-train the interrupted model, the instruction is like this “python train_PL_G2D.py --train_all --batchsize 4 --pool max --data_dir D:\University\University-Release\train --gen 0 --seed 0 --name step1 --gpu 0 --lambda1 1.0 --lambda2 0.1 --easy_pos --resume”.
I set up 100 rounds, and the first time I trained until I was 50, and then when I continued to train, the training effect started again, and that's why

**Comment by ZelongZeng** [comment]
Hi DSAJLDNA, 
Sorry, my code is based on this project: https://github.com/layumi/University1652-Baseline/tree/master
I think I deleted or modified the parts of `--resume`. 
You can refer to the `resume section` of https://github.com/layumi/University1652-Baseline/blob/master/train.py to implement the related functionality. 

Best regards.

**Comment by ahworld22** [comment]
> 非常感谢您的回答，我已经解决了第二个问题，但是当我调试预训练模型时，出现了以下循环，我想问一下是什么情况。这是顺序“python test_PL_G2D.py --batchsize 8 --test_dir D：\University\University-Release\test --name G2D --gpu_ids 0 --low_altitude ”。我们使用比例尺：1 17119 -------test----------- 从 ./model\G2D\net_119_s0.pth 2 -> 3 加载模型：

Hello may i ask how you solve the problem with
if __name__ == '__main__':
            freeze_support()
            ...
